t/10_susedistribution.t

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/perl
+# Copyright @ SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+use testapi;
+use Test::MockModule;
+use susedistribution;
+
+my $testapi_mocked = Test::MockModule->new('testapi');
+$testapi_mocked->mock(type_string => sub { 'randompass' });
+my $susedistri = susedistribution->new();
+
+my $suse_mocked = Test::MockModule->new('susedistribution');
+$suse_mocked->noop(qw(send_key enter_cmd));
+# Increase counter for each wait_serial invocation
+my $wait_serial_hit = 0;
+$suse_mocked->mock(wait_serial => sub { $wait_serial_hit++; return 'Password:' });
+# Increase counter for each assert_screen invocation
+my $assert_screen_called = 0;
+$suse_mocked->mock(assert_screen => sub { $assert_screen_called++; return 1 });
+# Increase counter for each wait_still_screen invocation
+my $wait_still_screen_called = 0;
+$suse_mocked->mock(wait_still_screen => sub { $wait_still_screen_called++; return 1 });
+
+subtest 'handle_password_prompt on serial terminal checks wait_serial' => sub {
+    set_var('VIRTIO_CONSOLE', '1');
+    set_var('BACKEND', 'qemu');
+    $suse_mocked->mock('is_serial_terminal' => sub { 1 });
+    ok susedistribution::handle_password_prompt();
+    is $wait_serial_hit, 1;
+    is $assert_screen_called, 0;
+};
+
+subtest 'handle_password_prompt assert screen on when not serial_terminal' => sub {
+    set_var('BACKEND', 'qemu');
+    $suse_mocked->mock('is_serial_terminal' => sub { 0 });
+    ok susedistribution::handle_password_prompt();
+    is $wait_serial_hit, 1;
+    is $assert_screen_called, 1;
+};
+
+subtest 'script_sudo works when is not serial_terminal and wait>0' => sub {
+    set_var('BACKEND', 'qemu');
+    $suse_mocked->mock('is_serial_terminal' => sub { return 0 });
```

Comment:
Lines like this could be written as
```suggestion
    $suse_mocked->mock(is_serial_terminal => 0);
```

Reply:
i thought i had changed that. checking

Reply:
Somehow i revert that part. i have changed it. I will submit in the morning

Reply:
Done *ready for review*